### PR TITLE
Implemented code refactor and created TaskExecutionHandler and TaskAc…

### PR DESCRIPTION
**Pull request by BHAVESUMEDH**
Implemented code refactor and created TaskExecutionHandler and TaskActionHandler

### CODE REVIEW

File: java/src/main/java/com/codurance/training/tasks/TaskList.java [hunk 1]
```diff
@@ -1,35 +1,29 @@
...
-    public TaskList(BufferedReader reader, PrintWriter writer) {
-        this.in = reader;
-        this.out = writer;
+    public TaskList() {
+        //default constructor without parameter
     }
 
     public void run() {
+        BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
+        PrintWriter out = new PrintWriter(System.out);
```
Comment (AbhilashZemoso) [inline java/src/main/java/com/codurance/training/tasks/TaskList.java]: these should be getting from main()
cannot create instance here

File: java/src/main/java/com/codurance/training/tasks/actions/TaskActionHandlerImpl.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+public class TaskActionHandlerImpl implements TaskActionHandler {
+
+    private final PrintWriter out = new PrintWriter(System.out);
```
Comment (AbhilashZemoso) [inline java/src/main/java/com/codurance/training/tasks/actions/TaskActionHandlerImpl.java]: this should not be created
Use the instance from main()

File: java/src/main/java/com/codurance/training/tasks/actions/TaskActionHandlerImpl.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+    private static long lastId = 0;
+
+    private static final Map<String, List<Task>> tasks = new LinkedHashMap<>();
```
Comment (AbhilashZemoso) [inline java/src/main/java/com/codurance/training/tasks/actions/TaskActionHandlerImpl.java]: move initialisation to constructor

File: java/src/main/java/com/codurance/training/tasks/actions/TaskActionHandlerImpl.java [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    }
+
+    public void error(String command) {
+        out.printf("I don't know what the command \"%s\" is.", command);
+        out.println();
+    }
+
+    public static long nextId() {
+        return ++lastId;
+    }
```
Comment (AbhilashZemoso) [inline java/src/main/java/com/codurance/training/tasks/actions/TaskActionHandlerImpl.java]: class needs to be segregated as Single Responsibility is not achieved

File: java/src/main/java/com/codurance/training/tasks/handler/TaskExecutionHandlerImpl.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
+package com.codurance.training.tasks.handler;
+
+import com.codurance.training.tasks.actions.TaskActionHandlerImpl;
+
+public class TaskExecutionHandlerImpl implements TaskExecutionHandler{
+    private final TaskActionHandlerImpl taskActionHandlerImpl = new TaskActionHandlerImpl();
```
Comment (AbhilashZemoso) [inline java/src/main/java/com/codurance/training/tasks/handler/TaskExecutionHandlerImpl.java]: move initialisation to constructor

**Review by AbhilashZemoso — CHANGES_REQUESTED** [review]
work on comments

File: java/src/main/java/com/codurance/training/tasks/actions/TaskActionHandlerImpl.java (hunk 2 above)
Comment (BHAVESUMEDH) [reply java/src/main/java/com/codurance/training/tasks/actions/TaskActionHandlerImpl.java]: Done

File: java/src/main/java/com/codurance/training/tasks/actions/TaskActionHandlerImpl.java (hunk 3 above)
Comment (BHAVESUMEDH) [reply java/src/main/java/com/codurance/training/tasks/actions/TaskActionHandlerImpl.java]: Done

File: java/src/main/java/com/codurance/training/tasks/actions/TaskActionHandlerImpl.java (hunk 4 above)
Comment (BHAVESUMEDH) [reply java/src/main/java/com/codurance/training/tasks/actions/TaskActionHandlerImpl.java]: Done

**Comment by BHAVESUMEDH** [comment]
> work on comments

DOne

File: java/src/main/java/com/codurance/training/tasks/handler/TaskExecutionHandlerImpl.java (hunk 5 above)
Comment (BHAVESUMEDH) [reply java/src/main/java/com/codurance/training/tasks/handler/TaskExecutionHandlerImpl.java]: done

**Pull request merged**